repo/blob/s3/s3_storage.go

```diff
@@ -142,13 +142,18 @@ func (s *s3Storage) putBlob(ctx context.Context, b blob.ID, data blob.Bytes) (ve
 		return versionMetadata{}, errors.Wrap(err, "AddReader")
 	}
 
-	storageClass := s.storageConfig.getStorageClassForBlobID(b)
-
-	uploadInfo, err := s.cli.PutObject(ctx, s.BucketName, s.getObjectNameString(b), throttled, int64(data.Length()), minio.PutObjectOptions{
+	putOpts := minio.PutObjectOptions{
 		ContentType:    "application/x-kopia",
 		SendContentMd5: atomic.LoadInt32(&s.sendMD5) > 0,
-		StorageClass:   storageClass,
-	})
+	}
+
+	blobOpts := s.storageConfig.forBlobID(b)
+	if blobOpts != nil {
+		putOpts.StorageClass = blobOpts.StorageClass
+		putOpts.RetainUntilDate = blobOpts.RetainUntilDate
```

Comment:
The option / property for the put blob operation is `RetainUntilDate`. However, the configuration parameter should be expressed as a duration, perhaps named `RetentionPeriod`. Then, the retain until date would be something like: `putOpts.RetainUntilDate = now().Add(opts.RetentionPeriod)1